src/pages/portfolio/ui/assets-table.tsx

```diff
@@ -0,0 +1,305 @@
+import { Table } from '@penumbra-zone/ui/Table';
+import { Card } from '@penumbra-zone/ui/Card';
+import { Text } from '@penumbra-zone/ui/Text';
+import { ArrowDownRight, ArrowUpRight } from 'lucide-react';
+import { Density } from '@penumbra-zone/ui/Density';
+import { useBalances } from '@/shared/api/balances';
+import { useAssets } from '@/shared/api/assets';
+import { observer } from 'mobx-react-lite';
+import { Skeleton } from '@/shared/ui/skeleton';
+import { ValueViewComponent } from '@penumbra-zone/ui/ValueView';
+import { connectionStore } from '@/shared/model/connection';
+import {
+  getMetadataFromBalancesResponse,
+  getBalanceView,
+} from '@penumbra-zone/getters/balances-response';
+import { BalancesResponse } from '@penumbra-zone/protobuf/penumbra/view/v1/view_pb';
+import { AssetIcon } from '@penumbra-zone/ui/AssetIcon';
+import { getDisplayDenomExponent } from '@penumbra-zone/getters/metadata';
+import { formatAmount } from '@penumbra-zone/types/amount';
+import { useRouter } from 'next/navigation';
+import { Button } from '@penumbra-zone/ui/Button';
+
+const LoadingState = () => {
+  return (
+    <div className='p-6'>
+      <Card>
+        <Text color='text.primary'>Assets</Text>
+
+        {/* Asset distribution bar skeleton */}
+        <div className='w-full h-2 mt-4 mb-6'>
+          <Skeleton />
+        </div>
+
+        {/* Legend skeleton */}
+        <div className='flex flex-wrap gap-4 mb-6'>
+          {Array.from({ length: 6 }).map((_, i) => (
+            <div key={i} className='flex items-center gap-2'>
+              <div className='w-3 h-3'>
+                <Skeleton />
+              </div>
+              <div className='w-20 h-4'>
+                <Skeleton />
+              </div>
+            </div>
+          ))}
+        </div>
+
+        <Density compact>
+          <Table>
+            <Table.Thead>
+              <Table.Tr>
+                <Table.Th>Asset</Table.Th>
+                <Table.Th>Balance</Table.Th>
+                <Table.Th>Price</Table.Th>
+                <Table.Th>Value</Table.Th>
+                <Table.Th>Actions</Table.Th>
+              </Table.Tr>
+            </Table.Thead>
+            <Table.Tbody>
+              {Array.from({ length: 6 }).map((_, i) => (
+                <Table.Tr key={i}>
+                  <Table.Td>
+                    <div className='flex items-center gap-2'>
+                      <div className='w-8 h-8'>
```

Comment:
todo: not-so-blocking but would be nice to have skeleton sizes be aligned with Figma correctly. Most of the sizes here differ from designs. For example, AssetIcon is `size-6` (here `size-8`), texts are `h-5` (here `h-4`), action buttons are `size-8` (here `size-5`)

Reply:
Good catch, yeah, I'll make sure to check those.